src/Compilers/Core/Portable/Compilation/TypeInfo.cs

```diff
@@ -17,12 +17,24 @@ public struct TypeInfo : IEquatable<TypeInfo>
         /// </summary>
         public ITypeSymbol Type { get; }
 
+        /// <summary>
+        /// The inferred nullability of the expression represented by the syntax node.
+        /// </summary>
+        public Nullability Nullability { get; }
```

Comment:
I don't like the naming here.  If type field is called `ITypeSymbol Type` it feels like this should be `Nullability TypeNullability`.  That way you can easily track what this is referring to.  This would also apply to 'ConvertedType' and 'ConvertedTypeNullability'. it basically augments the other field with that name.